scripts/silo/rpcsetup.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python3
+
+import json  # noqa
+import re
+from dataclasses import dataclass
+from typing import Any, Mapping
+
+import click
+
+"""Convenience script for setting up silos in a dev environment.
+
+Prints devserver commands with the necessary environment variables.
+"""
+
+
+@dataclass(frozen=True)
+class RegionConfig:
+    number: int
+    port: int
+    api_token: str
+
+    @property
+    def name(self) -> str:
+        return f"devregion{self.number}"
+
+    @property
+    def bind(self) -> str:
+        return f"localhost:{self.port}"
+
+    def get_env_repr(self) -> Mapping[str, Any]:
+        return {
+            "name": self.name,
+            "id": self.number,
+            "address": f"http://{self.bind}/",
+            "category": "MULTI_TENANT",
+            "api_token": self.api_token,
+        }
+
+
+def format_env_vars(env_vars: Mapping[str, str]) -> str:
+    def format_env_var(name: str, value: str) -> str:
+        value = re.sub("([\"$'\\\\])", "\\\\\\1", value)  # https://xkcd.com/1638/
```

Comment:
I find it helpful to use raw strings (`r'blah\1'`) for patterns as the escaping gets really silly.

Reply:
Yep, I generally do that with regexes. I could be wrong, but I think the first one actually *can't* be written as a raw string because it includes both `"` and `'` (therefore at least one needs to use a non-raw `\`).

I almost wrote the second one as `r"\\\1"` but then I thought... eh, it's just a helper script, and how often do I have an excuse to summon an elder backslash? 🤡

Reply:
shlex.quote is kinda designed for this if I'm understanding the code correclty